jupyter_elfinder/elfinder.py

```diff
@@ -543,39 +537,34 @@ def __zipdl(self) -> None:
         pass
 
     def __file(self) -> None:
+        self._http_header["Content-type"] = "text/html"
         target = self._request.get(API_TARGET)
         if not target:
-            self._response[R_ERROR] = "Invalid parameters"
+            self._response["__text"] = "Invalid parameters"
             return
 
         download = self._request.get(API_DOWNLOAD)
         cur_file = self._find(target)
 
         if not cur_file or not os.path.exists(cur_file) or os.path.isdir(cur_file):
-            self.http_status_code = 404
-            self.http_header["Content-type"] = "text/html"
-            self.http_response = "File not found"
+            self._http_status_code = 404
+            self._response["__text"] = "File not found"
             return
 
         if not self._is_allowed(cur_file, "read"):
-            self.http_status_code = 403
-            self.http_header["Content-type"] = "text/html"
-            self.http_response = "Access denied"
+            self._http_status_code = 403
+            self._response["__text"] = "Access denied"
             return
 
         if os.path.islink(cur_file):
             cur_file = self._read_link(cur_file)
-            if not cur_file or os.path.isdir(cur_file):
-                self.http_status_code = 404
-                self.http_header["Content-type"] = "text/html"
-                self.http_response = "File not found"
-                return
-            if not self._is_allowed(
-                os.path.dirname(cur_file), "read"
-            ) or not self._is_allowed(cur_file, "read"):
-                self.http_status_code = 403
-                self.http_header["Content-type"] = "text/html"
-                self.http_response = "Access denied"
+            if (
+                not cur_file
+                or not self._is_allowed(os.path.dirname(cur_file), "read")
+                or not self._is_allowed(cur_file, "read")
+            ):
+                self._http_status_code = 403
```

Comment:
Should we add content type header?

Reply:
See https://github.com/imjoy-team/jupyter-elfinder/pull/81#discussion_r399739995